ui/app/src/modules/Content/Authoring/NewContentDialog.tsx

```diff
@@ -0,0 +1,447 @@
+import Checkbox from '@material-ui/core/Checkbox';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Box from '@material-ui/core/Box';
+import Grid from '@material-ui/core/Grid';
+import CloseRoundedIcon from '@material-ui/icons/CloseRounded';
+import InsertDriveFileOutlinedIcon from '@material-ui/icons/InsertDriveFileOutlined';
+import { palette } from '../../../styles/theme';
+import { fetchLegacyContentTypes } from '../../../services/content';
+import DialogHeader from '../../../components/DialogHeader';
+import NewContentCard from './NewContentCard';
+import NewContentSelect from './NewContentSelect';
+import SearchBar from '../../../components/SearchBar';
+import ContentTypesFilter from './ContentTypesFilter';
+import { Item } from '../../../models/Item';
+import {
+  useDebouncedInput,
+  useSelection,
+  useSpreadState,
+  useStateResource
+} from '../../../utils/hooks';
+import DialogBody from '../../../components/DialogBody';
+import DialogFooter from '../../../components/DialogFooter';
+import EmbeddedLegacyEditors from '../../Preview/EmbeddedLegacyEditors';
+import Typography from '@material-ui/core/Typography';
+import { SuspenseWithEmptyState } from '../../../components/SystemStatus/Suspencified';
+import { APIError, EntityState } from '../../../models/GlobalState';
+import { LegacyFormConfig } from '../../../models/ContentType';
+import { Resource } from '../../../models/Resource';
+
+const translations = defineMessages({
+  title: {
+    id: 'newContentDialog.title',
+    defaultMessage: 'Create Content'
+  },
+  subtitle: {
+    id: 'newContentDialog.subtitle',
+    defaultMessage: 'Choose a content type template for your new content item.'
+  },
+  previewImage: {
+    id: 'previewImage.alt',
+    defaultMessage: 'preview'
+  },
+  compactInput: {
+    id: 'compactInput.label',
+    defaultMessage: 'Compact'
+  },
+  contentTypeAll: {
+    id: 'contentTypeAll.type',
+    defaultMessage: 'all'
+  },
+  contentTypeAllLabel: {
+    id: 'contentTypeAll.label',
+    defaultMessage: 'Show all types'
+  },
+  contentTypePage: {
+    id: 'contentTypePage.type',
+    defaultMessage: 'page'
+  },
+  contentTypePageLabel: {
+    id: 'contentTypePage.label',
+    defaultMessage: 'Pages only'
+  },
+  contentTypeComponent: {
+    id: 'contentTypeComponent.type',
+    defaultMessage: 'component'
+  },
+  contentTypeComponentLabel: {
+    id: 'contentTypeComponent.label',
+    defaultMessage: 'Components only'
+  },
+  contentTypeQuickCreate: {
+    id: 'contentTypeQuickCreate.type',
+    defaultMessage: 'quickCreate'
+  },
+  contentTypeQuickCreateLabel: {
+    id: 'contentTypeQuickCreate.label',
+    defaultMessage: 'Quick create only'
+  },
+  contentTypeFavorite: {
+    id: 'contentTypeFavorite.type',
+    defaultMessage: 'favorite'
+  },
+  contentTypeFavoriteLabel: {
+    id: 'contentTypeFavorite.label',
+    defaultMessage: 'Favorites only'
```

Comment:
- Keys needn't be translated - please remove
- The id root should be consistent e.g. `newContentDialog.a`, `newContentDialog.b`, `newContentDialog.c`
- `previewImage` is not an appropriate alt. Those images don't actually need an alt.